### PR TITLE
Support latest react and additional props #9

### PR DESCRIPTION
**Pull request by kof**

### CODE REVIEW

**Comment by kof** [comment]
Updated the PR.

**Comment by sapegin** [comment]
@alexkuz Could you please review this PR? Or maybe give @kof push rights to the repo if he wants to maintain the project and if you don’t mind?

**Comment by alexkuz** [comment]
@sapegin I'm sorry I'm not following this issue with proper attention lately.
I don't mind giving rights to @kof if he's up to it, but as for this issue - I'm still not convinced it's necessary, unless there are some performance issues (that should be solved by other means anyway).

**Comment by kof** [comment]
I think there should be a better way to achieve what I wanted. Clear perf issue is to create markdown-it instances all the time. Ultimately it will require an incompatible public api change.

File: src/index.js [hunk 1]
```diff
@@ -130,24 +130,33 @@ function mdReactFactory(options={}) {
...
     }
 
-    const children = tree.map(
+    let children = tree.map(
       (branch, idx) => Array.isArray(branch) ?
         iterateTree(branch, level + 1, idx) :
         branch
     );
 
+    // Don't pass empty children array to void elements like img,
```
Comment (sapegin) [inline src/index.js:154]: At least this part is necessary for React 0.14+.

Comment (kof) [reply src/index.js:154]: yep

**Comment by ajayapte** [comment]
I am seeing this error with 0.3.0 when there is an image component in md.  Example - 
readme = '# Some test \n This is a test \n ' +
        'Image alt text of mine'

Can someone accept this PR if code change is correct?

**Comment by kof** [comment]
I have got commit rights, will do soon.

**Comment by smm-telus** [comment]
Any progress on this?

**Comment by kof** [comment]
merged

**Comment by smm-telus** [comment]
Are there plans to publish these changes to NPM?

**Comment by kof** [comment]
done

**Pull request merged**